index/scorch/segment/segment.go

```diff
@@ -111,10 +111,28 @@ type Location interface {
 // postings or other indexed values.
 type DocumentFieldTermVisitable interface {
 	VisitDocumentFieldTerms(localDocNum uint64, fields []string,
-		visitor index.DocumentFieldTermVisitor) error
+		visitor index.DocumentFieldTermVisitor, optional DocVisitState) (DocVisitState, error)
 
 	// VisitableDocValueFields implementation should return
 	// the list of fields which are document value persisted and
 	// therefore visitable by the above VisitDocumentFieldTerms method.
 	VisitableDocValueFields() ([]string, error)
 }
+
+type DocVisitState interface {
+	State() *FieldDocValueState
+	SetState(*FieldDocValueState)
+}
+
+// FieldDocValueState represents the state details,
+// which intents to save the redundant dvCache preparations
+type FieldDocValueState struct {
+	DvFieldsAllPersisted bool
+	DvFieldsPending      []string
+	DvCachePrepared      bool
+	DvSegment            DocumentFieldTermVisitable
+}
+
+func (fdvs *FieldDocValueState) CurrentSegment() DocumentFieldTermVisitable {
+	return fdvs.DvSegment
+}
```

Comment:
The term "segment" kinda already has overloaded meaning in scorch, and this isn't returning a segment?  Perhaps the caller should access the DvSegment field directly?